### PR TITLE
VDS-2889: Upload file for import

### PR DESCRIPTION
**Pull request by ilyawzrd**
Description: 

--

QA-test: VDS-2901

Demo-test: VDS-2902

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/VirtoCommerce.CatalogExportImport_1.0.0-pr-16-e4051fe5.zip

### CODE REVIEW

**Review by asvishnyakov — CHANGES_REQUESTED** [review]
Feature flag is missed?

File: src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        [Name("Category Outer Id")]
+        public string CategoryOuterId { get; set; }
+
+        [Required]
+        [Name("Product Type")]
+        public string ProductType { get; set; }
+
+        [Optional]
+        [Name("Priority")]
+        public string Priority { get; set; }
```
Comment (asvishnyakov) [inline src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs]: `int?` (nullable int)
https://github.com/VirtoCommerce/vc-module-catalog/blob/1a3aa89755eefcb4b5dda4e7d001f5fda3507552/src/VirtoCommerce.CatalogModule.Core/Model/CatalogProduct.cs#L97

File: src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        [Name("Package Type")]
+        public string PackageType { get; set; }
+
+        [Optional]
+        [Name("Max Quantity")]
+        public string MaxQuantity { get; set; }
+
+        [Optional]
+        [Name("Min Quantity")]
+        public string MinQuantity { get; set; }
```
Comment (asvishnyakov) [inline src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs]: `int?`

File: src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        [BooleanFalseValues("no", "false")]
+        public bool? TrackInventory { get; set; }
+
+        [Optional]
+        [Name("Package Type")]
+        public string PackageType { get; set; }
+
+        [Optional]
+        [Name("Max Quantity")]
+        public string MaxQuantity { get; set; }
```
Comment (asvishnyakov) [inline src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs]: `int?`

File: src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        [Name("Measure Unit")]
+        public string MeasureUnit { get; set; }
+
+        [Optional]
+        [Name("Weight Unit")]
+        public string WeightUnit { get; set; }
+
+        [Optional]
+        [Name("Weight")]
+        public string Weight { get; set; }
```
Comment (asvishnyakov) [inline src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs]: `decimal?`

File: src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+        [Optional]
+        [Name("Lenght")]
+        public string Lenght { get; set; }
```
Comment (asvishnyakov) [inline src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs]: `decimal?`

File: src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+        [Optional]
+        [Name("Width")]
+        public string Width { get; set; }
```
Comment (asvishnyakov) [inline src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs]: `decimal?`

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs (hunk 1 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs]: done

File: src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs (hunk 2 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs]: done

File: src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs (hunk 3 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs]: done

File: src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs (hunk 6 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs]: done

File: src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs (hunk 5 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs]: done

File: src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs (hunk 4 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.CatalogExportImportModule.Core/Models/CsvPhysicalProduct.cs]: done

**Review by asvishnyakov — APPROVED** [review]

File: src/VirtoCommerce.CatalogExportImportModule.Web/Scripts/module.js [hunk 7]
```diff
@@ -7,13 +7,15 @@ if (AppDependencies !== undefined) {
...
+angular.module(moduleName, ['ui.grid.autoFitColumns']).run(['virtoCommerce.featureManagerSubscriber', 'virtoCommerce.catalogModule.catalogImportService', 'virtoCommerce.catalogModule.catalogExportService', function (featureManagerSubscriber, catalogImportService, catalogExportService) {
+
     catalogImportService.register({
         name: 'Description import',
         description: 'Descriptions data import from CSV',
         icon: 'fa fa-list-alt',
         controller: 'virtoCommerce.catalogExportImportModule.fileUploadController',
-        template: 'Modules/$(VirtoCommerce.CatalogExportImport)/Scripts/blades/file-upload.tpl.html'
+        template: 'Modules/$(VirtoCommerce.CatalogExportImport)/Scripts/blades/file-upload.tpl.html',
+        predefinedDataType: { key: 'Descriptions', value: 'EditorialReview' }
```
Comment (trueboroda) [inline src/VirtoCommerce.CatalogExportImportModule.Web/Scripts/module.js:18]: maybe just 'dataType'?

File: src/VirtoCommerce.CatalogExportImportModule.Web/Scripts/blades/file-upload.tpl.html [hunk 8]
```diff
@@ -62,7 +62,8 @@
                             </div>
                             <div class="d-flex justify-content-end"  ng-if="!(csvMaxSizeError || fileTypeError)">
                                 <span class="btn btn-small btn-danger-outline" ng-class="{'mr-10': !internalCsvError}" ng-click="deleteUploadedItem()">{{ 'catalogExportImport.blades.file-upload.actions.delete' | translate }}</span>
-                                <span class="btn btn-small border-primary" ng-if="!internalCsvError" ng-click="showPreview()">{{ 'catalogExportImport.blades.file-upload.actions.preview' | translate }}</span>
+                                <span class="btn btn-small border-primary" ng-if="!internalCsvError && blade.dataType.value === 'EditorialReview'" ng-click="showPreview()">{{ 'catalogExportImport.blades.file-upload.actions.preview' | translate }}</span>
```
Comment (trueboroda) [inline src/VirtoCommerce.CatalogExportImportModule.Web/Scripts/blades/file-upload.tpl.html]: Use the real value comparison. For example 'PhisicalProduct'. And move these values to the frontend modules constans.

**Review by trueboroda — CHANGES_REQUESTED** [review]

File: src/VirtoCommerce.CatalogExportImportModule.Web/Scripts/module.js (hunk 7 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.CatalogExportImportModule.Web/Scripts/module.js:18]: We are setting blade.dataType inside fileUploadController

File: src/VirtoCommerce.CatalogExportImportModule.Web/Scripts/blades/file-upload.tpl.html (hunk 8 above)
Comment (ilyawzrd) [reply src/VirtoCommerce.CatalogExportImportModule.Web/Scripts/blades/file-upload.tpl.html]: done

**Review by trueboroda — APPROVED** [review]

**Pull request merged**